pkgs/development/compilers/dmd/default.nix

```diff
@@ -47,9 +126,11 @@ stdenv.mkDerivation rec {
 
       cd ../phobos
       mkdir $out/lib
-      ${let bits = if stdenv.is64bit then "64" else "32";
-            osname = if stdenv.isDarwin then "osx" else "linux"; in
-      "cp generated/${osname}/release/${bits}/libphobos2.a $out/lib"
+      ${
+          let bits = builtins.toString stdenv.hostPlatform.parsed.cpu.bits;
+          osname = if stdenv.isDarwin then "osx" else "linux";
```

Comment:
Thanks for making the other two fixes! Last thing would be using `hostPlatform` and not assuming Linux:
```nix
let
  bits = builtins.toString stdenv.hostPlatform.parsed.cpu.bits;
  osname = if stdenv.hostPlatform.isDarwin then "osx" else stdenv.hostPlatform.parsed.kernel.name;
  extension = if stdenv.hostPlatform.isDarwin then "a" else "{a,so}";
in ...
```